### PR TITLE
MINOR: update VerifiableProducer to send keys if configured and removed StreamsRepeatingKeyProducerService

### PR DESCRIPTION
**Pull request by bbejeck**
This PR does the following:
- Remove the `StreamsRepeatingIntegerKeyProducerService` and  the associated Java class 
- Add a parameter to `VerifiableProducer.java` to enable sending keys when specified
- Update the corresponding Python file `verifiable_producer.py` to support the new parameter.

Tested by kicking off all system tests https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1677/

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
ping @guozhangwang, @mjsax, @vvcephei 

\cc @ewencp for the `VerfiableProducer` changes.

**Comment by bbejeck** [comment]
Adding @dguy

File: tests/kafkatest/tests/streams/streams_standby_replica_test.py
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_standby_replica_test.py]: Why do we start the producer later compared to the driver it replaces?

**Review by mjsax — COMMENTED** [review]
What is the purpose of this PR? Reducing code redundancy?

Overall LGTM. One clarification question.

Can we trigger the system test before merging -- as `verifiable_producer` is updated, I would suggest to run the whole suite.

File: tests/kafkatest/tests/streams/streams_standby_replica_test.py
Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_standby_replica_test.py]: oversight during refactoring, I'll update this.

**Comment by bbejeck** [comment]
@mjsax, thanks for the comments.  Yes, PR is for reducing code redundancy, as previously the `StreamsRepeatingIntegerKeyProducerService` is merely a wrapper around a producer that uses keys.

As for running system tests, I kicked off the entire suite earlier today, and it's still running.

**Comment by bbejeck** [comment]
updated this

**Review by mjsax — APPROVED** [review]
LGTM -- assuming all tests pass.

**Comment by bbejeck** [comment]
First triggered system test build was hung (11 hours).  So kicked off a new build https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1681/

**Comment by bbejeck** [comment]
In [latest system test run](http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/2018-04-10--001.1523362701--bbejeck--MINOR_move_repeating_key_producer_from_service--0293a01/report.html) there were two failures 

1. `kafkatest.tests.core.transactions_test.TransactionsTest.test_transactions.failure_mode=clean_bounce.bounce_target=clients.check_order=False: FAIL: Message Copier failed to stop`

2. `kafkatest.tests.core.upgrade_test.TestUpgrade.test_upgrade.from_kafka_version=0.10.1.1.to_message_format_version=None.compression_types=.snappy.new_consumer=False: FAIL: Node ubuntu@worker11: did not stop within the specified timeout of 150 seconds`

But these seem unrelated to changes made in `VerifiableProducer.` 

At any rate, I've kicked off [another run of the entire system test suite](https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1685/)

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsStandByReplicaTest.java
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/tests/StreamsStandByReplicaTest.java]: Is this for debugging only or is it necessary for the test itself?

**Review by guozhangwang — COMMENTED** [review]
The change lgtm. Left a minor comment on the `print` call.

If the system test passed we can merge it.

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsStandByReplicaTest.java
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsStandByReplicaTest.java]: Ack. It's left over from debugging; I'll remove it.

**Comment by mjsax** [comment]
Retest this please.

@guozhangwang If the system test failure are unrelated, can we merge this anyway?

**Comment by bbejeck** [comment]
The same tests failed after a few attempts so I rebased and kicked off a new build earlier today https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1698/

**Comment by bbejeck** [comment]
[the latest build](http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/2018-04-12--001/report.html) was aborted after 13 hours.  There was one test failure `kafkatest.tests.connect.connect_distributed_test`, but the cause was `KeyboardInterrupt` so I think this was due to stopping the build.

I've kicked off another build  https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1700/

**Comment by bbejeck** [comment]
retest this please

**Comment by guozhangwang** [comment]
@bbejeck Could you rebase?

**Comment by bbejeck** [comment]
rebased this.

[the previous system test build](http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/2018-04-13--001/report.html) timed out, but there were no failures, 356 tests ran.

Kicking off a new system test build https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1701/

**Comment by guozhangwang** [comment]
The system test seem failed again, not sure if it the same you observed before:

```
17:20:15 [INFO:2018-04-16 17:20:15,620]: RunnerClient: kafkatest.tests.streams.streams_eos_test.StreamsEosTest.test_failure_and_recovery: FAIL: 
17:20:15 Traceback (most recent call last):
17:20:15   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/local/lib/python2.7/site-packages/ducktape-0.7.1-py2.7.egg/ducktape/tests/runner_client.py", line 132, in run
17:20:15     data = self.run_test()
17:20:15   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/local/lib/python2.7/site-packages/ducktape-0.7.1-py2.7.egg/ducktape/tests/runner_client.py", line 185, in run_test
17:20:15     return self.test_context.function(self.test)
17:20:15   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/tests/kafkatest/tests/streams/streams_eos_test.py", line 86, in test_failure_and_recovery
17:20:15     StreamsEosTestVerifyRunnerService(self.test_context, self.kafka))
17:20:15   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/tests/kafkatest/tests/streams/streams_eos_test.py", line 112, in run_failure_and_recovery
17:20:15     self.abort_streams(processor1, processor3, processor2)
17:20:15   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/tests/kafkatest/tests/streams/streams_eos_test.py", line 157, in abort_streams
17:20:15     self.wait_for_startup(monitor2, keep_alive_processor2)
17:20:15   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/tests/kafkatest/tests/streams/streams_eos_test.py", line 161, in wait_for_startup
17:20:15     self.wait_for(monitor, processor, "StateChange: REBALANCING -> RUNNING")
17:20:15   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/tests/kafkatest/tests/streams/streams_eos_test.py", line 167, in wait_for
17:20:15     err_msg=("Never saw output '%s' on " % output) + str(processor.node.account))
17:20:15   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/local/lib/python2.7/site-packages/ducktape-0.7.1-py2.7.egg/ducktape/cluster/remoteaccount.py", line 668, in wait_until
17:20:15     allow_fail=True) == 0, **kwargs)
17:20:15   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/local/lib/python2.7/site-packages/ducktape-0.7.1-py2.7.egg/ducktape/utils/util.py", line 34, in wait_until
17:20:15     time.sleep(backoff_sec)
17:20:15 KeyboardInterrupt
```

**Comment by bbejeck** [comment]
yes, that's the same error, the build sometimes takes too long. 
I've kicked off another one https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1720/

**Review by guozhangwang — APPROVED** [review]
I will merge this once the system test passed.

**Comment by guozhangwang** [comment]
The above system test still fails.

**Comment by bbejeck** [comment]
Yes, I saw that earlier.

One note is all of the `StreamsUpgradeTest `s are failing, but don't use the `VerifiableProducer` at all.  

The latest PR from @mjsax upgrading streams has all streams system tests passing.  I'm thinking maybe once that PR is merged I'll rebase and try again.  

WDYT?

**Comment by guozhangwang** [comment]
Re-triggered https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1734/

**Comment by bbejeck** [comment]
system-test build failed looks like its the JDK-7 related issue:
```00:53:15 ==> worker5: Package oracle-java7-set-default is not available, but is referred to by another package.
00:53:15 ==> worker5: This may mean that the package is missing, has been obsoleted, or
00:53:15 ==> worker5: is only available from another source
00:53:15 ==> worker5: However the following packages replace it:
00:53:15 ==> worker5:   oracle-java8-set-default
00:53:15 ==> worker5: 
00:53:15 ==> worker5: E: Unable to locate package oracle-java7-installer
00:53:15 ==> worker5: E: Package 'oracle-java7-set-default' has no installation candidate
00:53:15 ==> worker5: + echo 'ERROR: JDK install failed'
00:53:15 ==> worker5: ERROR: JDK install failed
00:53:15 ==> worker5: + exit 1
```

**Comment by bbejeck** [comment]
needed to rebase. Kicked off new system test https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1735/

**Comment by guozhangwang** [comment]
Failures in 1735 is not related, I'm going to merge it now.

**Comment by guozhangwang** [comment]
Thanks @bbejeck !

**Pull request merged**
